dipy/tracking/utils.py

```diff
@@ -484,6 +489,45 @@ def target(streamlines, target_mask, affine, include=True):
             yield sl
 
 
+def near_roi(streamlines, target_mask, affine=None, tol=0):
+    """
+    Provide filtering criteria for a set of streamlines based on whether they
+    fall within a tolerance distance from an ROI
+
+    Parameters
+    ----------
+    streamlines: list
+        A sequence of streamlines. Each streamline should be a (N, 3) array,
+        where N is the length of the streamline.
+    target_mask: ndarray
+        A mask used as a target. Non-zero values are considered to be within
+        the target region.
+    affine: ndarray
+        Affine transformation from voxels to streamlines. Default: identity.
+    tol: float
+        Distance (in the units of the streamlines, usually mm). If any
+        coordinate in the streamline is within this distance from any voxel in
+        the ROI, the filtering criterion is set to True for this streamlin,
+        otherwise False. Default: 0
+
+    Returns
+    -------
+    Boolean array. Set to True in each streamline
+    """
+    if affine is None:
+        affine = np.eye(4)
+    roi_coords = np.array(np.where(target_mask)).T
+    x_roi_coords = apply_affine(affine, roi_coords)
+    out = np.zeros(len(streamlines), dtype=bool)
+    for ii, sl in enumerate(streamlines):
+        for coord in sl:
```

Comment:
The "first" is in quotes because maybe you have already written many cythonized functions and I am not aware of it. Definitely not ironic. This is a critical functionality and it will be a bottleneck if we are not smart. I think Cythonizing those intersections with ROIs is a great idea and will not add any problems on maintaining the code. These are small functions. If you see that the speed improvement is not considerable then we talk again. But I think you will be surprised from the speedup.